### ISSUE TITLE
0.17.0 Regression: Some legal queries fail

### ISSUE THREAD

**Issue by smackesey** [tree-sitter/node-tree-sitter#69]
After upgrading to tree-sitter 0.17.0, some of my queries started crashing my code. The queries are part of a fairly sizable grammar, so I won't post a ton of information unless it's requested. But one query that started crashing was this:

```
new TSQuery(MyLanguage, '(header_1 (citekey))')
```

Attempting to define this query causes an immediate crash with this error message:

```
Assertion failed: (final_step_indices.size > 0), function ts_query__analyze_patterns, file ../vendor/tree-sitter/lib/src/./query.c, line 1258.
error Command failed with signal "SIGABRT".
```

I've confirmed that on reverting to 0.16.x, everything works again.

I looked a bit at the source and I think the code block the error comes from is only supposed to run if the query is unmatchable. But this query is not unmatchable, so if I'm correct that's another bug.

**Comment by maxbrunsfeld** [comment]
Ok, I'll see if I can debug it without the specific grammar. If that fails, is the grammar closed source, or would you be able to share it with me to help debug?

**Comment by smackesey** [comment]
Not closed source. Attaching
[grammar.js.txt](https://github.com/tree-sitter/node-tree-sitter/files/5345388/grammar.js.txt)
[scanner.cc.txt](https://github.com/tree-sitter/node-tree-sitter/files/5345391/scanner.cc.txt)

 grammar.js and scanner.cc here in case you need them. Thanks!

**Comment by maxbrunsfeld** [comment]
Ok, I think I fixed the bug that caused this. When you have time, could you try to build the latest commit of Tree-sitter, and run your full query?

To build:

```sh
git clone https://github.com/tree-sitter/tree-sitter
cd tree-sitter
cargo build --release  # need to have Rust installed
```

Then, to test your query, run this in your grammar directory:

```sh
$TREE_SITTER_DIRECTORY/target/release/tree-sitter query  $YOUR_QUERY_FILE  $SOME_EXAMPLE_MEMEWIKI_FILE
```

**Comment by maxbrunsfeld** [comment]
I published node-tree-sitter 0.17.1. I'm going to close this out since your query loaded without error. Let me know if you see further problems!
